### PR TITLE
Surround page by a ScrolledComposite

### PR DESCRIPTION
**Pull request by jessyec-s**
This patch adds vertical and horizontal scrollbars to the page in order to accommodate small screen sizes

### CODE REVIEW

File: application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/ChartAndPopupTableUI.java [hunk 1]
```diff
@@ -223,6 +234,19 @@ public void handleEvent(Event event) {
 		scText.setExpandHorizontal(true);
 		scText.setExpandVertical(true);
 
+		scPageContainer.addListener( SWT.Resize, new Listener() {
```
Comment (aptmac) [inline application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/ChartAndPopupTableUI.java]: whitespace before `SWT.Resize`

File: application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/ChartAndPopupTableUI.java [hunk 2]
```diff
@@ -223,6 +234,19 @@ public void handleEvent(Event event) {
...
+			@Override
+			public void handleEvent(Event event) {
+				int width = filterBar.getClientArea().width;
+				int height = timelineCanvas.getClientArea().height + filterBar.getClientArea().height
+						+ scChart.getClientArea().height;
+				if (width > 0 && height > 0) {
+					 scPageContainer.setMinSize(scPageContainer.computeSize( width, height ) );
```
Comment (aptmac) [inline application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/ChartAndPopupTableUI.java]: excess whitespace around the `width, height`

**Review by aptmac — COMMENTED** [review]
Nice! Overall I think this approach looks good, and thanks for digging into the colouring issues with the scrolled composite.

I think it's nearly there, I just have one nit with regards to the resizing. I think it may have something to do with the minimum size being set, but when the page is resized smaller it used to decrease the width of the filter bar, and the page could be squished to a smaller size before the page would get folded away. Now it looks like the scrolled composite gets a certain size set, and when resized smaller it keeps that dimension and adds the scroll bars. This was harder than I thought to put into words, but here are a couple gifs to show.

Current: horizontally resizing the page smaller causes the page to shrink until the filter bar components are "squished" together and then it folds behind the page boundary
drag-previous

PR: the scrolled composite looks to maintain the current size, and then uses the scrollbars to maintain that size even when offscreen.
drag-after

Also not directly related to this PR (but kind of is..?) I was curious as to what some general UX guidelines were for minimum supported application dimensions. Both [Microsoft](https://docs.microsoft.com/en-us/windows/win32/uxguide/top-violations) [0] and [GNOME](https://developer.gnome.org/hig/stable/display-compatibility.html.en) [1] recommend 1024 pixels as a minimum width, and the threads page currently manages to fit comfortably on my screen (at 96 DPI, screenshot below). Also seeing as how the default width for JMC is 1400px [2] then resizing smaller is dependant on the user, so I think this approach should be enough to address this concern. If somehow the scrolled composite would fold away like the first gif I showed and use scrollbars for sizes smaller than that, then it would handle both 1024px size without modification and screen sizes smaller than this with the scroll bars .. what are your thoughts?
1024-width

[0] https://docs.microsoft.com/en-us/windows/win32/uxguide/top-violations
[1] https://developer.gnome.org/hig/stable/display-compatibility.html.en
[2] https://github.com/openjdk/jmc/blob/a07f3a28e65993909f6281ca5617f0ecc2b152a9/application/org.openjdk.jmc.rcp.application/src/main/java/org/openjdk/jmc/rcp/application/ApplicationWorkbenchWindowAdvisor.java#L59

File: application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/ChartAndPopupTableUI.java (hunk 1 above)
Comment (jessyec-s) [reply application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/ChartAndPopupTableUI.java]: fixed https://github.com/aptmac/jmc/pull/45/commits/caa3552fda290a50a95516ddb195b1de19914796

File: application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/ChartAndPopupTableUI.java (hunk 2 above)
Comment (jessyec-s) [reply application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/ChartAndPopupTableUI.java]: fixed https://github.com/aptmac/jmc/pull/45/commits/caa3552fda290a50a95516ddb195b1de19914796

**Comment by jessyec-s** [comment]
I have modified the minimum width and height of the scrolled composite in https://github.com/aptmac/jmc/pull/45/commits/caa3552fda290a50a95516ddb195b1de19914796.

Now the width and height are computed by referencing the minimum widths and heights of the sc's sub components

**Pull request merged**